ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAPP-512 Savings summary page

## Description of change
Adds savings summary page displaying any savings 
Adds change and remove actions 
Change actions routes user to the savings details page to edit details of saving 
Remove action takes user to confirm screen to confirm they would like to remove the saving. 
When saving is deleted user is routed back to the saving summary page if there are savings or the saving type page if there are no more savings 
Also adds question to user to add another savings account. If yes is selected they are routed to the savings type page, if no is selected they are routed to the premium bonds page (not yet implemented)

## Link to relevant ticket
[CRIMAPP-512](https://dsdmoj.atlassian.net/browse/CRIMAPP-512)
[CRIMAPP-513](https://dsdmoj.atlassian.net/browse/CRIMAPP-513) - to add confirm remove screen

## Notes for reviewer
This is the first of two PRs. The following CRIMAPP-512 PR will fix the display of the savings on the page to match design (currently just a list of the savings attributes are being displayed)

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

**Saving summary page** 
<img width="1046" alt="Screenshot 2024-02-23 at 14 30 12" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/1eb38844-9f0f-410e-a69a-8c2e1f1076b7">

**Validation error if user clicks save and continue without selecting option** 
<img width="1046" alt="Screenshot 2024-02-23 at 14 30 25" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/6464d3dd-b473-4197-972d-621636034831">

**Deletion of savings account routes to savings summary page if there are other savings** 
<img width="1046" alt="Screenshot 2024-02-23 at 14 33 08" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/09e50de1-f290-4acf-8c8f-0162137c7c22">
<img width="1046" alt="Screenshot 2024-02-23 at 14 35 53" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/6a9d4c22-4039-4035-b184-0cd26e1fbc66">

**Deletion of a savings account routes to savings type page if there are no other savings** 
<img width="1046" alt="Screenshot 2024-02-23 at 14 38 29" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/e36df9a2-951d-4cea-a43a-cab3ca21ff67">
<img width="1046" alt="Screenshot 2024-02-23 at 13 23 22" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/14618ccb-82cd-4d5d-8c1b-c9b33441b883">



## How to manually test the feature
Navigate to `steps/capital/which_savings_does_client_have`
Add a saving type and go to next screen
Fill in details of saving to add saving
Now you should be on the saving summary screen 
You should see the saving account you just added

Test the following:
- That you can click the change link and go to the saving detail page and update the saving 
- That you can click remove and are taken to the confirm removal screen. Then test you can click no and go back to the savings summary. Also test you can click yes and delete the saving
- That when you delete a saving and there are other savings, you are taken to the savings summary page
- That when you delete a saving and there are no other savings, you are taken to the premium bonds page (currently not implemented)
- On the saving summary page, click save and continue without answering the question of whether you want to add another saving account. Check you are shown a validation error 

Can also test that when you click save and continue whilst filling in saving details and navigate to the `steps/capital/clients_savings`, you are shown an incomplete tag for the saving with partial information filled

[CRIMAPP-512]: https://dsdmoj.atlassian.net/browse/CRIMAPP-512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-513]: https://dsdmoj.atlassian.net/browse/CRIMAPP-513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
